### ISSUE TITLE
ImportOrder with option=bottom should not allow non-static import after static import

### ISSUE THREAD

**Issue by julianhyde** [checkstyle/checkstyle#4981]
http://checkstyle.sourceforge.net/config_imports.html#ImportOrder 
http://checkstyle.sourceforge.net/property_types.html#importOrder

/var/tmp $ javac InputImportOrderStaticGroupOrderBottom_Negative.java

(no output)

/var/tmp $ cat -n InputImportOrderStaticGroupOrderBottom_Negative.java

```java
  1 package com.puppycrawl.tools.checkstyle.checks.imports.importorder;
  2
  3 import org.*;
  4
  5 import static java.lang.Math.PI;
  6 import static org.antlr.v4.runtime.Recognizer.EOF;
  7
  8 import java.util.Set;
  9
 10 public class InputImportOrderStaticGroupOrderBottom_Negative
 11 {
 12
 13 }
```

/var/tmp $ cat config.xml

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="charset" value="UTF-8"/>

  <module name="TreeWalker">
    <module name="ImportOrder">
      <property name="groups" value="org,java"/>
      <property name="option" value="bottom"/>
      <property name="sortStaticImportsAlphabetically" value="true"/>
    </module>
  </module>
</module>
```

/var/tmp $ java -jar checkstyle-X.XX-all.jar -c config.xml InputImportOrderStaticGroupOrderBottom_Negative.java

Gives no errors, but I would expect

```
InputImportOrderStaticGroupOrderBottom_Negative.java:8: Wrong order for 'java.util.Set' import.
```

**Comment by julianhyde** [comment]
There is a test case and fix in https://github.com/julianhyde/checkstyle/tree/import-bottom.

**Comment by julianhyde** [comment]
@romani Please review and approve.

**Comment by romani** [comment]
@julianhyde , sorry for delay, thank for bringing it back to attention.
Issue is approved, please proceed with PR

**Comment by julianhyde** [comment]
See PR #5130

**Comment by rnveach** [comment]
@julianhyde 
> InputImportOrderStaticGroupOrderBottom_Negative.java 
Gives no errors

I get an error which regresses all the way back to version 7, but it is not the violation you are asking for.
````
$ cat TestClass.java
package com.puppycrawl.tools.checkstyle.checks.imports.importorder;

import org.*;

import static java.lang.Math.PI;
import static org.antlr.v4.runtime.Recognizer.EOF;

import java.util.Set;

public class InputImportOrderStaticGroupOrderBottom_Negative {
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
    <module name="ImportOrder">
      <property name="groups" value="org,java"/>
      <property name="ordered" value="true"/>
      <property name="separated" value="true"/>
      <property name="option" value="bottom"/>
    </module>
    </module>
</module>

$ java -jar checkstyle-8.2-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:6: Wrong order for 'org.antlr.v4.runtime.Recognizer.EOF' import. [ImportOrder]
Audit done.
Checkstyle ends with 1 errors.
````

When I fix up this error I get the error you are looking for:
````
$ cat TestClass.java
package com.puppycrawl.tools.checkstyle.checks.imports.importorder;

import org.*;

import static org.antlr.v4.runtime.Recognizer.EOF;

import static java.lang.Math.PI;

import java.util.Set;

public class InputImportOrderStaticGroupOrderBottom_Negative {
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
    <module name="ImportOrder">
      <property name="groups" value="org,java"/>
      <property name="ordered" value="true"/>
      <property name="separated" value="true"/>
      <property name="option" value="bottom"/>
    </module>
    </module>
</module>

$ java -jar checkstyle-8.2-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:9: Wrong order for 'java.util.Set' import. [ImportOrder]
Audit done.
Checkstyle ends with 1 errors.
````

Am I missing something? Can you fix up your issue so it is repeatable?

**Comment by julianhyde** [comment]
My mistake. I have changed the contents of config.xml at the top of this case. Now checkstyle gives 0 errors without the fix, 1 error with the fix, as desired.

**Comment by romani** [comment]
fix is merged.
Thanks a lot to all participants.
